integration_tests/test_remote.py

```diff
@@ -0,0 +1,92 @@
+from git_wrapper.repo import GitRepo
+
+
+def test_prune_tags(repo_root, clone_repo_root):
+    repo = GitRepo(repo_root)
+    cloned_repo = GitRepo.clone(f"file://{repo_root}", clone_repo_root)
+
+    cloned_repo.tag.create("test_tag", "master")
+    assert "test_tag" in cloned_repo.tag.names()
+
+    cloned_repo.tag.push("test_tag", "origin")
+    assert "test_tag" in repo.tag.names()
+
+    cloned_repo.git.push("origin", "--delete", "test_tag")
+    assert "test_tag" not in repo.tag.names()
+    assert "test_tag" in cloned_repo.tag.names()
+    cloned_repo.remote.fetch("origin", prune=True, prune_tags=True)
+    assert "test_tag" not in cloned_repo.tag.names()
+
+
+def test_no_prune(repo_root, clone_repo_root):
+    repo = GitRepo(repo_root)
+    cloned_repo = GitRepo.clone(f"file://{repo_root}", clone_repo_root)
+
+    cloned_repo.tag.create("test_tag", "master")
+    assert "test_tag" in cloned_repo.tag.names()
+
+    cloned_repo.tag.push("test_tag", "origin")
+    assert "test_tag" in repo.tag.names()
+
+    cloned_repo.remote.fetch("origin")
```

Comment:
I think we need to delete on the remote before fetching, to prove that fetch didn't delete anything.